### ISSUE TITLE
Episode 01 - transition to jupyter

### ISSUE THREAD

**Issue by isaacWpark** [datacarpentry/python-ecology-lesson#502]
given that this episode is the only one that uses python directly from the command line, it might be worth making where the transition to jupyter should occur more clear - As it stands, it's unclear whether the entire episode should be taught in the command-line python instance, or whether we should trnasition to jupyter after demonstrating how to run a python script from the command line.

**Comment by yasharEM** [comment]
Agreed, in the workshop we had recently this was a source of major confusion for participants. I think it's better to start from command-line (or Anaconda prompt for example) and demonstrate the process of running a python script from there in detail. The current lesson is not focused on this part and assume participants have prior knowledge about command line and bash. Then, when everybody understands how it can be done, proceed to jupyter and continue the lesson.

**Comment by YasharMonf** [comment]
Agreed! This was an issue in our recent workshop and caused some confusion. The current lesson structure assumes that participants have prior knowledge about command-line or bash, which in many cases is not true. In my opinion, the lesson should be started with pure command-line (anaconda prompt for example) focus and demonstrate the process of running a script completely inside CM. Then, we can proceed to jupyter which can be valuable for learners.

**Comment by yasharEM** [comment]
Agreed! This was an issue in our recent workshop and caused some confusion. The current lesson structure assumes that participants have prior knowledge about command-line or bash, which in many cases is not true. In my opinion, the lesson should be started with pure command-line (anaconda prompt for example) focus and demonstrate the process of running a script completely inside CM. Then, we can proceed to jupyter which can be valuable for learners.

**Comment by eug-xu** [comment]
To add onto both previous points; I just taught this episode and here are the additions/modifications I had to make to the curriculum to make this lesson make sense: 

1. I think that the lesson plan, beyond being clearer in its movement from command line to jupyter, needs to include an introduction to the command line because most learners taking this lesson also don't have background on the command line.
2. The lesson plan has learners interact with the command line, .py files, and .ipynb files. I found it very necessary to explain the difference between a python notebook from a .py file. I also think all the carpentries intro python episodes need to include somewhere in the start some guidance on explaining cells, the kernel, and key keyboard shortcuts (I think the Data Analysis and Visualization with Python for Social Scientists Introduction to Python episode contains a lot of this information)
3. The lesson also wants the instructor to jump from command line to jupyter, then back to command line to demonstrate why printing is important, then back to jupyter, all within the first ~20 minutes of the lesson. I know that live coding is part of the carpentries pedagogy but I ended up just verbally explaining the difference between printing and reviewing the value of a variable because I think switching back and forth like that was going to be way too disorienting for the learners.

**Comment by jas58** [comment]
Thanks @eug-xu for your explanation.  I'm reviewing for a workshop this week, and I while I've attended several python workshops, I cannot make heads or tails of this lesson.  There's one mention lost in setup about Shift+Return keys.  It's so buried.  It needs to be it's own sentence or callout box.  Along the lines of 
:::::::::::::::::::::::::callout
Shift Key plus Return Key are the magic you've been looking for!  Once you've typed the command or equation into the bottom line of the jupyter notebook, instead of using Enter, you must press both the shift and the return/enter key together every time you want to run that line of code.
:::::::::::::::::::::::::

But now I'm on the next box and completely lost.  I cannot get the anaconda prompt window to respond to my keyboard though there are commands and it loaded all of the miniconda stuff.  It took me a minute to notice the very subtle "python" and "bash" lost in a corner of the command visuals in the lessons, too.  But Gitbash was not any better with the first call for a script.  I was told to load python from the Microsoft Store.  Or disable the shortcut about aliases (?????).

![Screenshot of listed attempts.](https://github.com/user-attachments/assets/f55cd932-020d-46b1-a994-c28dc6403395)


Wait, now that I'm reading it again, the bash box is not mean to be coded by the learners (right?) only demo'ed by an instructor who has a functional python with that code already loaded to call?  If the latter, that is not explicit at all in the text.  

Would that need explicit call outs to 

::::::::::::::::::::::::::::::callout
"Watch the instructor's screen. You will not perform any commands until variables are introduced in the next episode."  
::::::::::::::::::::::::::::::

Is that correct?

**Comment by jas58** [comment]
Eug-Xu's comment 2. is also valid: what elements of the Social Scientist version can be brought in for instructors not already well-versed in it to borrow from?
